peer.go

```diff
@@ -131,9 +135,23 @@ func (l *PeerList) Get(prevSelected map[string]struct{}) (*Peer, error) {
 		peer = l.choosePeer(prevSelected, false /* avoidHost */)
 	}
 	if peer == nil {
+		return nil, ErrNoNewPeers
+	}
+	return peer, nil
+}
+
+// Get returns a peer from the peer list, or nil if none can be found,
+// will avoid previously selected peers if possible.
+func (l *PeerList) Get(prevSelected map[string]struct{}) (*Peer, error) {
+	peer, err := l.GetNew(prevSelected)
+	if err != nil && err != ErrNoNewPeers {
```

Comment:
@prashantv ah nice spot, it could race without the lock and the peer it would have returned got removed.